resources/web/docs_js/index.js

```diff
@@ -170,6 +170,34 @@ function init_toc(lang_strings) {
     });
 }
 
+function highlight_otp() {
+  let visibileHeadings = []
+  const observer = new IntersectionObserver(entries => {
+    entries.forEach(entry => {
+      const id = entry.target.getAttribute('id');
+      let element = document.querySelector(`#sticky_content #this_page a[href="#${id}"]`);
+      let itemId = $(element).parent().attr('id')
+      if (entry.intersectionRatio > 0){
+        visibileHeadings.push(itemId);
+      } else {
+        const position = visibileHeadings.indexOf(itemId);
+        visibileHeadings.splice(position, position + 1)
+      }
+      if (visibileHeadings.length > 0) {
+        visibileHeadings.sort()
```

Comment:
[I'm assuming ids are numerical]

At this point the `itemId`s will be strings and the sort will start failing when the id hits double digits,

```
const arr = [];
for (let i = 0; i < 11; i++) arr.push(`${i}`)
arr.sort();
// ['0', '1', '10', '2', '3', '4', '5', '6', '7', '8', '9']
```

need to parse them to numbers first,

```
let itemId = parseInt($(element).parent().attr('id'), 10)
```

Reply:
Ahhh thanks @chandlerprall! 

The `itemId`s are strings (for example `otp-text-1`, `otp-text-2`).  

https://github.com/elastic/docs/blob/99eab4ec17a3f9248b2a30b31d242a7575e94c6d/resources/web/docs_js/index.js#L60

I don't _think_ we want to use just a numbered `id` for list items without the context of `otp-text-` (but maybe it would be ok?).

Reply:
This seems to work, but there is probably a more elegant solution. 😝 

```js
    entries.forEach(entry => {
      const id = entry.target.getAttribute('id');
      let element = document.querySelector(`#sticky_content #this_page a[href="#${id}"]`);
      let itemId = $(element).parent().attr('id')
      let itemNumber = parseInt(itemId.match(/otp-text-(\d*)/)[1], 10)
      if (!itemNumber) return
      if (entry.intersectionRatio > 0){
        visibileHeadings.push(itemNumber);
      } else {
        const position = visibileHeadings.indexOf(itemNumber);
        visibileHeadings.splice(position, 1)
      }
      if (visibileHeadings.length > 0) {
        visibileHeadings.sort((a, b) => a - b)
        // Remove existing active classes
        $('a.active').removeClass("active");
        // Add active class to the first visible heading
        $('#otp-text-' + visibileHeadings[0] + ' > a').addClass('active')
      }
    })
```

Reply:
Ah, that's still workable; it can parse out the numeric id with a regex instead,

```
const itemId = parseInt(
  $(element).parent().attr('id').match(/\d+/)[0],
  10
);
```

If that creates too much work in other places that wants to use the full string id, the call to `.sort()` can do the parsing work. A bit more CPU overhead, but as long as we don't have pages with 100s of headings it shouldn't be noticable,

```
visibleHeadings.sort((a, b) => {
  const aId = parseInt(
    a.match(/\d+/)[0],
    10
  );
  const bId = parseInt(
    a.match(/\d+/)[0],
    10
  );
  if (aId < bId) return -1;
  if (aId > bId) return 1;
  return 0;
});
```